CONTRIBUTING.md

```diff
@@ -1,9 +1,40 @@
 # Contributing to Atomic Reactor
 
 If you're reading this, it means that you want to contribute to Atomic Reactor. Great! Here are some tips to make sure your pull request gets accepted.
-Note: there are quite a few tools mentioned through this tutorial. To get them all, including Atomic Reactor dependencies, you can run `pip install -r requirements-devel.txt`.
+Note: there are quite a few tools mentioned through this tutorial. To get them all, including Atomic Reactor dependencies, see the [Environment preparation](#environment-preparation) section.
+
 Also note: while this seems to be an awfully long text, chances are you adhere to most of it just because you're a great developer. And don't worry! Us, core devs, must send code through pull requests, too (well, assuming it isn't an obvious oneliner).
 
+## Environment preparation
+
+While Atomic Reactor depends on the `docker-py` python package, one of its dependencies (docker-squash) requires the `docker` package to be installed.  Although docker and docker-py have different names in pypi, they are just different versions of the same package, i.e., they are installed under the same namespace. Hence, when we install one of these packages after the other, pip will overwrite files from the first installed package with files from the second one.
+
+For now, Atomic Reactor does not support the `docker` python package, so we want to keep the `docker-py` installation in our environment. For that, you must install docker-squash (which pulls docker) before docker-py, so the former will be overwritten by the latter.
+
+Note that just changing the order of the lists in the requirements file [does not guarantee the packages will be installed in that order](https://pip.pypa.io/en/stable/reference/pip_install/#installation-order).
+
+To make sure `docker` will get pulled before docker-py (and then overwritten by it), we can install our dependencies with
+
+```
+python setup.py develop
+pip install -r requirements-devel.txt
+```
+
+If you not not want to run `python setup.py develop`, make sure you install docker-squash before using the requirements list
+
+```
+pip install docker-squash
+pip install -r requirements-devel.txt
+```
+
+Finally, you will also need to install the OSBS client
+
+```
+pip install git+https://github.com/projectatomic/osbs-client
```

Comment:
We can define this in `requirements-devel.txt`. Also, let's add dockpulp in there too. It's not optional, but unit tests for plugins that require dockpulp will be skipped.

Reply:
When I install `dockpulp` in a python 3 env
`tests/plugins/test_koji_promote.py::TestKojiPromote::test_koji_promote_success[True-True-True-True-v1+v2-True-2-False-registry_digests1-False-10485760-None]` fails with [1].

In a python 2 env, the same test fails with [1], but only when `dockpulp` **is not** installed.

In the CI environment, we only install `dockpulp` for python 2 environments.

[1]:
```
_____________________________________________ TestKojiPromote.test_koji_promote_success[True-True-True-True-v1+v2-True-2-False-registry_digests1-False-10485760-None] _____________________________________________

self = <tests.plugins.test_koji_promote.TestKojiPromote object at 0x7fc63c342160>, tmpdir = local('/tmp/pytest-of-athos/pytest-35/test_koji_promote_success_True0'), apis = 'v1+v2', docker_registry = True
pulp_registries = 2, pulp_supports_schema2 = False, registry_digests = {'v1': 'sha256:1000000000000000000000000000001d', 'v2': 'sha256:2000000000000000000000000000001d'}, metadata_only = False
blocksize = 10485760, target = None, os_env = None, has_config = True, is_autorebuild = True, tag_later = True, reactor_config_map = True

    @pytest.mark.parametrize(('apis',
                              'docker_registry',
                              'pulp_registries',
                              'pulp_supports_schema2',
                              'registry_digests',
                              'metadata_only',
                              'blocksize',
                              'target'), [
        ('v1-only',
         False,
         1,
         False,
         {'v1': 'sha256:1000000000000000000000000000001d'},
         False,
         None,
         'images-docker-candidate'),
    
        ('v1+v2',
         True,
         2,
         False,
         {'v1': 'sha256:1000000000000000000000000000001d',
          'v2': 'sha256:2000000000000000000000000000001d'},
         False,
         10485760,
         None),
    
        ('v2-only',
         True,
         1,
         True,
         {'v2': 'sha256:2000000000000000000000000000001d'},
         True,
         None,
         None),
    
        ('v1+v2',
         True,
         0,
         False,
         {'v1': 'sha256:1000000000000000000000000000001d',
          'v2': 'sha256:2000000000000000000000000000001d'},
         False,
         10485760,
         None),
    ])
    @pytest.mark.parametrize(('has_config', 'is_autorebuild'), [
        (True,
         True),
        (False,
         False),
    ])
    @pytest.mark.parametrize('tag_later', (True, False))
    def test_koji_promote_success(self, tmpdir, apis, docker_registry,
                                  pulp_registries, pulp_supports_schema2,
                                  registry_digests,
                                  metadata_only, blocksize,
                                  target, os_env, has_config, is_autorebuild,
                                  tag_later, reactor_config_map):  # noqa
        session = MockedClientSession('')
        # When target is provided koji build will always be tagged,
        # either by koji_promote or koji_tag_build.
        (flexmock(session)
            .should_call('tagBuild')
            .with_args('images-candidate', '123')
            .times(1 if target else 0))
    
        component = 'component'
        name = 'ns/name'
        version = '1.0'
        release = '1'
    
        if has_config and not docker_registry:
            # Not a valid combination
            has_config = False
    
        tasker, workflow = mock_environment(tmpdir,
                                            session=session,
                                            name=name,
                                            component=component,
                                            version=version,
                                            release=release,
                                            docker_registry=docker_registry,
                                            pulp_registries=pulp_registries,
                                            pulp_supports_schema2=pulp_supports_schema2,
                                            registry_digests=registry_digests,
                                            blocksize=blocksize,
                                            has_config=has_config)
        workflow.prebuild_results[CheckAndSetRebuildPlugin.key] = is_autorebuild
        runner = create_runner(tasker, workflow, metadata_only=metadata_only,
                               blocksize=blocksize, target=target,
                               tag_later=tag_later, reactor_config_map=reactor_config_map)
        result = runner.run()
    
        # Look at koji_tag_build result for determining which plugin
        # performing koji build taggging
        if tag_later and target:
            assert result[KojiTagBuildPlugin.key] == 'images-candidate'
        else:
            assert result.get(KojiTagBuildPlugin.key) is None
    
        data = session.metadata
        if metadata_only:
            mdonly = set()
        else:
            mdonly = set(['metadata_only'])
    
        assert set(data.keys()) == set([
            'metadata_version',
            'build',
            'buildroots',
            'output',
        ])
    
        assert data['metadata_version'] in ['0', 0]
    
        build = data['build']
        assert isinstance(build, dict)
    
        buildroots = data['buildroots']
        assert isinstance(buildroots, list)
        assert len(buildroots) > 0
    
        output_files = data['output']
        assert isinstance(output_files, list)
    
        assert set(build.keys()) == set([
            'name',
            'version',
            'release',
            'source',
            'start_time',
            'end_time',
            'owner',
            'extra',          # optional but always supplied
            'metadata_only',  # only when True
        ]) - mdonly
    
        assert build['name'] == component
        assert build['version'] == version
        assert build['release'] == release
        assert build['source'] == 'git://hostname/path#123456'
        start_time = build['start_time']
        assert isinstance(start_time, int) and start_time
        end_time = build['end_time']
        assert isinstance(end_time, int) and end_time
        if metadata_only:
            assert isinstance(build['metadata_only'], bool)
            assert build['metadata_only']
    
        extra = build['extra']
        assert isinstance(extra, dict)
        assert 'image' in extra
        image = extra['image']
        assert isinstance(image, dict)
        assert 'autorebuild' in image
        autorebuild = image['autorebuild']
        assert isinstance(autorebuild, bool)
        assert autorebuild == is_autorebuild
    
        for buildroot in buildroots:
            self.validate_buildroot(buildroot)
    
            # Unique within buildroots in this metadata
            assert len([b for b in buildroots
                        if b['id'] == buildroot['id']]) == 1
    
        if docker_registry:
            # v2 support
            if 'v1' in registry_digests and 'v2' in registry_digests:
                if not pulp_registries:
                    expect_digest = (2,)
                else:
                    if pulp_supports_schema2:
                        expect_digest = (1, 2)
                    else:
                        if sys.version_info.major > 2:
                            expect_digest = (2,)
                        else:
                            expect_digest = (1,)
            elif 'v1' in registry_digests:
                expect_digest = (1,)
            elif 'v2' in registry_digests:
                expect_digest = (2,)
        else:
            # no v2 support
            expect_digest = ()
    
        for output in output_files:
            self.validate_output(output, metadata_only, has_config,
                                 expect_digest=expect_digest,
>                                expected_digests=registry_digests, docker_registry=docker_registry)

tests/plugins/test_koji_promote.py:1250: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tests.plugins.test_koji_promote.TestKojiPromote object at 0x7fc63c342160>, output = {'arch': 'x86_64', 'buildroot_id': 1, 'checksum': '20439f79e4e9dc95be34b21029221f80', 'checksum_type': 'md5', ...}
metadata_only = False, has_config = True, expect_digest = (2,), expected_digests = {'v1': 'sha256:1000000000000000000000000000001d', 'v2': 'sha256:2000000000000000000000000000001d'}, docker_registry = True

    def validate_output(self, output, metadata_only, has_config,
                        expect_digest, expected_digests, docker_registry):
        if metadata_only:
            mdonly = set()
        else:
            mdonly = set(['metadata_only'])
    
        assert isinstance(output, dict)
        assert 'buildroot_id' in output
        assert 'filename' in output
        assert output['filename']
        assert is_string_type(output['filename'])
        assert 'filesize' in output
        assert int(output['filesize']) > 0 or metadata_only
        assert 'arch' in output
        assert output['arch']
        assert is_string_type(output['arch'])
        assert 'checksum' in output
        assert output['checksum']
        assert is_string_type(output['checksum'])
        assert 'checksum_type' in output
        assert output['checksum_type'] == 'md5'
        assert is_string_type(output['checksum_type'])
        assert 'type' in output
        if output['type'] == 'log':
            assert set(output.keys()) == set([
                'buildroot_id',
                'filename',
                'filesize',
                'arch',
                'checksum',
                'checksum_type',
                'type',
                'metadata_only',  # only when True
            ]) - mdonly
            assert output['arch'] == 'noarch'
        else:
            assert set(output.keys()) == set([
                'buildroot_id',
                'filename',
                'filesize',
                'arch',
                'checksum',
                'checksum_type',
                'type',
                'components',
                'extra',
                'metadata_only',  # only when True
            ]) - mdonly
            assert output['type'] == 'docker-image'
            assert is_string_type(output['arch'])
            assert output['arch'] != 'noarch'
            assert output['arch'] in output['filename']
            self.check_components(output['components'])
    
            extra = output['extra']
            assert isinstance(extra, dict)
            assert set(extra.keys()) == set([
                'image',
                'docker',
            ])
    
            image = extra['image']
            assert isinstance(image, dict)
            assert set(image.keys()) == set([
                'arch',
            ])
    
            assert image['arch'] == output['arch']  # what else?
    
            assert 'docker' in extra
            docker = extra['docker']
            assert isinstance(docker, dict)
            expected_keys_set = set([
                'parent_id',
                'id',
                'repositories',
                'digests',
                'layer_sizes',
                'tags',
            ])
            if has_config:
                expected_keys_set.add('config')
            if not docker_registry:
                expected_keys_set.remove('digests')
            assert set(docker.keys()) == expected_keys_set
    
            assert is_string_type(docker['parent_id'])
            assert is_string_type(docker['id'])
            repositories = docker['repositories']
            assert isinstance(repositories, list)
            repositories_digest = list(filter(lambda repo: '@sha256' in repo, repositories))
            repositories_tag = list(filter(lambda repo: '@sha256' not in repo, repositories))
    
            assert len(repositories_tag) == 1
            assert len(repositories_digest) == len(expect_digest)
    
            # check for duplicates
            assert sorted(repositories_tag) == sorted(set(repositories_tag))
            assert sorted(repositories_digest) == sorted(set(repositories_digest))
    
            for repository in repositories_tag:
                assert is_string_type(repository)
                image = ImageName.parse(repository)
                assert image.registry
                assert image.namespace
                assert image.repo
                assert image.tag and image.tag != 'latest'
    
            # These fake digests have the schema number as the first digit
            schemas = []
            for pullspec in repositories_digest:
                # drop image name (before '@')
                _, digest = pullspec.split('@', 1)
                # drop algorithm name (before 'sha256:')
                _, hashval = digest.split(':', 1)
                schemas.append(int(hashval[0]))
    
            # Expect schema 1 digest if only one digest expected, or
            # schema 1 and schema 2 if two digests are expected
            for schema in expect_digest:
>               assert schema in schemas
E               assert 2 in [1]

tests/plugins/test_koji_promote.py:896: AssertionError
-------------------------------------------------------------------------------------------------- Captured log ---------------------------------------------------------------------------------------------------
source.py                   68 DEBUG    workdir is '/tmp/tmpk9v366_9'
source.py                   75 DEBUG    source path is '/tmp/tmpk9v366_9/path'
config.py                   21 DEBUG    Trying paths: ['/home/athos/.docker/config.json', '/home/athos/.dockercfg']
config.py                   28 DEBUG    No config file found
config.py                   21 DEBUG    Trying paths: ['/home/athos/.docker/config.json', '/home/athos/.dockercfg']
config.py                   28 DEBUG    No config file found
source.py                   68 DEBUG    workdir is '/tmp/tmpa44do4_f'
source.py                   75 DEBUG    source path is '/tmp/tmpa44do4_f/docker-hello-world'
plugin.py                  494 INFO     initializing runner of exit plugins
plugin.py                  130 DEBUG    loading plugins from dir '/home/athos/src/atomic-reactor/atomic_reactor/plugins'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_remove_worker_metadata.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_koji_promote.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_import_image.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_pyrpkg_fetch_artefacts.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_verify_media_types.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_resolve_module_compose.py'
plugin.py                  145 WARNING  can't load module '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_resolve_module_compose.py': ModuleNotFoundError("No module named 'flatpak_module_tools'",)
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_change_from_in_df.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_fetch_worker_metadata.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/input_osv3.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_labels_in_df.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_tag_by_labels.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_compress.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_resolve_composes.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_rpmqa.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_pulp_tag.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_store_logs_to_file.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_flatpak_create_oci.py'
plugin.py                  145 WARNING  can't load module '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_flatpak_create_oci.py': ModuleNotFoundError("No module named 'flatpak_module_tools'",)
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_distribution_scope.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_tag_from_config.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_yum_repo.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/input_env.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_bump_release.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_sendmail.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_store_metadata_in_osv3.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_check_and_set_rebuild.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_remove_built_image.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/__init__.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/build_orchestrate_build.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_squash.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/build_imagebuilder.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/input_path.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_pulp_publish.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_pull_base_image.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_delete_from_registry.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_fetch_maven_artifacts.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_group_manifests.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_dockerfile.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_koji_upload.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_check_and_set_platforms.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_pulp_pull.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_tag_and_push.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/build_docker_api.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_pulp_sync.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_yum_repo_by_url.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_koji_parent.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_inject_parent_image.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_inject_yum_repo.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_koji_tag_build.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_reactor_config.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_koji.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_filesystem.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_help.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_koji_import.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_flatpak_create_dockerfile.py'
plugin.py                  145 WARNING  can't load module '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_flatpak_create_dockerfile.py': ModuleNotFoundError("No module named 'flatpak_module_tools'",)
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_compare_components.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_push_to_pulp.py'
plugin.py                  140 DEBUG    load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_tests_for_image.py'
plugin.py                  231 DEBUG    running plugin 'koji_promote'
plugin.py                  395 INFO     running plugin instance with args: '{'kojihub': '', 'url': '/', 'blocksize': 10485760}'
koji_util.py               116 INFO     Using Kerberos for Koji authentication
util.py                   1191 DEBUG    base image unable to be inspected
parser.py                  118 DEBUG    Setting inherited parent image ENV vars: {}
parser.py                  409 DEBUG    label value: 'BZComponent=component com.redhat.component=component'
parser.py                  416 DEBUG    new label 'BZComponent'='component'
parser.py                  416 DEBUG    new label 'com.redhat.component'='component'
parser.py                  409 DEBUG    label value: 'Version=1.0 version=1.0'
parser.py                  416 DEBUG    new label 'Version'='1.0'
parser.py                  416 DEBUG    new label 'version'='1.0'
parser.py                  409 DEBUG    label value: 'Release=1 release=1'
parser.py                  416 DEBUG    new label 'Release'='1'
parser.py                  416 DEBUG    new label 'release'='1'
parser.py                  420 DEBUG    instructions: {'BZComponent': 'component', 'com.redhat.component': 'component', 'Version': '1.0', 'version': '1.0', 'Release': '1', 'release': '1'}
exit_koji_promote.py       554 INFO     build is isolated: False
util.py                    523 DEBUG    md5sum: 2e3cd9cc1ff19e104f7e8042df45f5d3
util.py                    523 DEBUG    md5sum: 868d06093d8b87b97501978ab24c0904
util.py                    523 DEBUG    md5sum: 20439f79e4e9dc95be34b21029221f80
exit_koji_promote.py       646 DEBUG    uploading '/tmp/build-1dwj6xe10.log' to 'koji-promote/1539275934.5297158.LhwaNUMr' as 'openshift-final.log'
exit_koji_promote.py       651 DEBUG    using blocksize 10485760
exit_koji_promote.py       657 DEBUG    uploaded 'koji-promote/1539275934.5297158.LhwaNUMr/openshift-final.log'
exit_koji_promote.py       646 DEBUG    uploading '/tmp/docker-build-12vd9l1bj.log' to 'koji-promote/1539275934.5297158.LhwaNUMr' as 'build.log'
exit_koji_promote.py       651 DEBUG    using blocksize 10485760
exit_koji_promote.py       657 DEBUG    uploaded 'koji-promote/1539275934.5297158.LhwaNUMr/build.log'
exit_koji_promote.py       646 DEBUG    uploading '/tmp/pytest-of-athos/pytest-35/test_koji_promote_success_True0/image.tar.xz' to 'koji-promote/1539275934.5297158.LhwaNUMr' as 'docker-image-123456imageid.x86_64.tar.xz'
exit_koji_promote.py       651 DEBUG    using blocksize 10485760
exit_koji_promote.py       657 DEBUG    uploaded 'koji-promote/1539275934.5297158.LhwaNUMr/docker-image-123456imageid.x86_64.tar.xz'
exit_koji_promote.py       705 DEBUG    Build information: {
    "id": "123"
}
plugin.py                  288 DEBUG    plugin 'koji_promote' finished in 0s
---------------------------------------------------------------------------------------------- Captured stderr call -----------------------------------------------------------------------------------------------
2018-10-11 18:38:54,476 platform:- - atomic_reactor.source - DEBUG - workdir is '/tmp/tmpk9v366_9'
2018-10-11 18:38:54,476 platform:- - atomic_reactor.source - DEBUG - source path is '/tmp/tmpk9v366_9/path'
2018-10-11 18:38:54,479 platform:- - atomic_reactor.source - DEBUG - workdir is '/tmp/tmpa44do4_f'
2018-10-11 18:38:54,479 platform:- - atomic_reactor.source - DEBUG - source path is '/tmp/tmpa44do4_f/docker-hello-world'
2018-10-11 18:38:54,479 platform:- - atomic_reactor.plugin - INFO - initializing runner of exit plugins
2018-10-11 18:38:54,480 platform:- - atomic_reactor.plugin - DEBUG - loading plugins from dir '/home/athos/src/atomic-reactor/atomic_reactor/plugins'
2018-10-11 18:38:54,480 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_remove_worker_metadata.py'
2018-10-11 18:38:54,480 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_koji_promote.py'
2018-10-11 18:38:54,481 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_import_image.py'
2018-10-11 18:38:54,481 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_pyrpkg_fetch_artefacts.py'
2018-10-11 18:38:54,482 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_verify_media_types.py'
2018-10-11 18:38:54,482 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_resolve_module_compose.py'
2018-10-11 18:38:54,483 platform:- - atomic_reactor.plugin - WARNING - can't load module '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_resolve_module_compose.py': ModuleNotFoundError("No module named 'flatpak_module_tools'",)
2018-10-11 18:38:54,483 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_change_from_in_df.py'
2018-10-11 18:38:54,483 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_fetch_worker_metadata.py'
2018-10-11 18:38:54,483 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/input_osv3.py'
2018-10-11 18:38:54,483 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_labels_in_df.py'
2018-10-11 18:38:54,484 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_tag_by_labels.py'
2018-10-11 18:38:54,484 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_compress.py'
2018-10-11 18:38:54,486 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_resolve_composes.py'
2018-10-11 18:38:54,487 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_rpmqa.py'
2018-10-11 18:38:54,487 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_pulp_tag.py'
2018-10-11 18:38:54,488 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_store_logs_to_file.py'
2018-10-11 18:38:54,488 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_flatpak_create_oci.py'
2018-10-11 18:38:54,490 platform:- - atomic_reactor.plugin - WARNING - can't load module '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_flatpak_create_oci.py': ModuleNotFoundError("No module named 'flatpak_module_tools'",)
2018-10-11 18:38:54,490 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_distribution_scope.py'
2018-10-11 18:38:54,490 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_tag_from_config.py'
2018-10-11 18:38:54,490 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_yum_repo.py'
2018-10-11 18:38:54,491 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/input_env.py'
2018-10-11 18:38:54,491 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_bump_release.py'
2018-10-11 18:38:54,491 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_sendmail.py'
2018-10-11 18:38:54,497 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_store_metadata_in_osv3.py'
2018-10-11 18:38:54,498 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_check_and_set_rebuild.py'
2018-10-11 18:38:54,498 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_remove_built_image.py'
2018-10-11 18:38:54,498 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/__init__.py'
2018-10-11 18:38:54,498 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/build_orchestrate_build.py'
2018-10-11 18:38:54,499 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_squash.py'
2018-10-11 18:38:54,514 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/build_imagebuilder.py'
2018-10-11 18:38:54,515 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/input_path.py'
2018-10-11 18:38:54,515 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_pulp_publish.py'
2018-10-11 18:38:54,515 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_pull_base_image.py'
2018-10-11 18:38:54,515 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_delete_from_registry.py'
2018-10-11 18:38:54,516 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_fetch_maven_artifacts.py'
2018-10-11 18:38:54,517 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_group_manifests.py'
2018-10-11 18:38:54,517 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_dockerfile.py'
2018-10-11 18:38:54,517 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_koji_upload.py'
2018-10-11 18:38:54,518 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_check_and_set_platforms.py'
2018-10-11 18:38:54,518 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_pulp_pull.py'
2018-10-11 18:38:54,519 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_tag_and_push.py'
2018-10-11 18:38:54,519 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/build_docker_api.py'
2018-10-11 18:38:54,519 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_pulp_sync.py'
2018-10-11 18:38:54,519 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_yum_repo_by_url.py'
2018-10-11 18:38:54,520 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_koji_parent.py'
2018-10-11 18:38:54,521 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_inject_parent_image.py'
2018-10-11 18:38:54,521 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_inject_yum_repo.py'
2018-10-11 18:38:54,521 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_koji_tag_build.py'
2018-10-11 18:38:54,521 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_reactor_config.py'
2018-10-11 18:38:54,522 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_koji.py'
2018-10-11 18:38:54,522 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_filesystem.py'
2018-10-11 18:38:54,523 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_add_help.py'
2018-10-11 18:38:54,523 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/exit_koji_import.py'
2018-10-11 18:38:54,524 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_flatpak_create_dockerfile.py'
2018-10-11 18:38:54,525 platform:- - atomic_reactor.plugin - WARNING - can't load module '/home/athos/src/atomic-reactor/atomic_reactor/plugins/pre_flatpak_create_dockerfile.py': ModuleNotFoundError("No module named 'flatpak_module_tools'",)
2018-10-11 18:38:54,525 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_compare_components.py'
2018-10-11 18:38:54,526 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/post_push_to_pulp.py'
2018-10-11 18:38:54,526 platform:- - atomic_reactor.plugin - DEBUG - load file '/home/athos/src/atomic-reactor/atomic_reactor/plugins/prepub_tests_for_image.py'
2018-10-11 18:38:54,526 platform:- - atomic_reactor.plugin - DEBUG - running plugin 'koji_promote'
2018-10-11 18:38:54,526 platform:- - atomic_reactor.plugin - INFO - running plugin instance with args: '{'kojihub': '', 'url': '/', 'blocksize': 10485760}'
2018-10-11 18:38:54,527 platform:- - atomic_reactor.koji_util - INFO - Using Kerberos for Koji authentication
2018-10-11 18:38:54,527 platform:- - atomic_reactor.util - DEBUG - base image unable to be inspected
2018-10-11 18:38:54,528 platform:- - atomic_reactor.plugins.koji_promote - INFO - build is isolated: False
2018-10-11 18:38:54,529 platform:- - atomic_reactor.util - DEBUG - md5sum: 2e3cd9cc1ff19e104f7e8042df45f5d3
2018-10-11 18:38:54,529 platform:- - atomic_reactor.util - DEBUG - md5sum: 868d06093d8b87b97501978ab24c0904
2018-10-11 18:38:54,529 platform:- - atomic_reactor.util - DEBUG - md5sum: 20439f79e4e9dc95be34b21029221f80
2018-10-11 18:38:54,529 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - uploading '/tmp/build-1dwj6xe10.log' to 'koji-promote/1539275934.5297158.LhwaNUMr' as 'openshift-final.log'
2018-10-11 18:38:54,529 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - using blocksize 10485760
2018-10-11 18:38:54,529 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - uploaded 'koji-promote/1539275934.5297158.LhwaNUMr/openshift-final.log'
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - uploading '/tmp/docker-build-12vd9l1bj.log' to 'koji-promote/1539275934.5297158.LhwaNUMr' as 'build.log'
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - using blocksize 10485760
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - uploaded 'koji-promote/1539275934.5297158.LhwaNUMr/build.log'
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - uploading '/tmp/pytest-of-athos/pytest-35/test_koji_promote_success_True0/image.tar.xz' to 'koji-promote/1539275934.5297158.LhwaNUMr' as 'docker-image-123456imageid.x86_64.tar.xz'
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - using blocksize 10485760
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - uploaded 'koji-promote/1539275934.5297158.LhwaNUMr/docker-image-123456imageid.x86_64.tar.xz'
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugins.koji_promote - DEBUG - Build information: {
    "id": "123"
}
2018-10-11 18:38:54,530 platform:- - atomic_reactor.plugin - DEBUG - plugin 'koji_promote' finished in 0s
```